jvcl/run/JvInterpreterParser.pas

```diff
@@ -501,10 +501,13 @@ function TokenTyp(const Token: string): TTokenKind;
 
         Point := False;
         IsScientificNotation := False;
-        for I := 1 to L1 do
+        for I := 2 to L1 do
```

Comment:
Isn't this skipping a character, even when there is no scientific notation?
Won't it have the consequence of allowing a.3?